src/main/java/org/jabref/gui/importer/actions/OpenDatabaseAction.java

```diff
@@ -195,15 +195,21 @@ private ParserResult loadDatabase(Path file) throws Exception {
 
         preferencesService.getFilePreferences().setWorkingDirectory(fileToLoad.getParent());
 
+        ParserResult result = null;
         if (BackupManager.backupFileDiffers(fileToLoad)) {
-            BackupUIManager.showRestoreBackupDialog(dialogService, fileToLoad);
+            var resultOpt = BackupUIManager.showRestoreBackupDialog(dialogService, fileToLoad, preferencesService);
+            if (resultOpt.isPresent()) {
+                result = resultOpt.get();
```

Comment:
You could simply with resultOpt.orElse(null)

Reply:
Ouch... 😖 😅